packages/tablev2/Table.tsx

```diff
@@ -0,0 +1,337 @@
+import * as React from "react";
+import { cx } from "emotion";
+import * as style from "./style";
+import { CheckboxCell, Draggable, HeaderCellContent } from "./Util";
+import { ColumnProps } from "./Column";
+import { ResetButton } from "../button";
+
+type ColumnChild<A, B> = React.ReactElement<ColumnProps<A, B>>;
+
+// TODO: make it possible to disable rows from being selected with a checkbox
+type TableProps<A, B = string, C extends B = B> = {
+  /**
+   * Table columns
+   */
+  children: Array<ColumnChild<A, B>> | ColumnChild<A, B>;
+
+  /**
+   * The ID and direction to initially sort by.
+   */
+  initialSorter?: { by: C; order?: "asc" | "desc" };
+
+  /**
+   * A callback invoked whenever the internal state of the table is changed.
+   * Have a look at the State type to see what's in there.
+   */
+  onStateChange?: (a: State) => void;
+
+  /**
+   * An object where the keys are the IDs of selected rows, and the value is a boolean
+   */
+  selectedRows?: {};
```

Comment:
how often is kommander using this?
i initially thought of solving this in userland by providing examples/helpers.

asking as it adds quite some complexity and currently rerenders every cell when something in the table updates.

Reply:
Kommander doesn't use this at all yet, but I think there are a few places in DC/OS that use this.

The re-render is definitely undesirable...maybe we could just figure out how to have selectable rows without baking it into the component, and add a Storybook example?

Curious to hear what you and other reviewers think about this.

Reply:
@pierrebeitz - Are you sure it's re-rendering every cell? I just tested it, and I'm only seeing the first column and the header row re-render on update.

Reply:
Oh, I'm sorry. I meant to say "every cell of the first column".